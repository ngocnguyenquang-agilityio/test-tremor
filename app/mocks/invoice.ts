export const INVOICE_DATA = {
  id: 230220,
  createdAt: "06/03/2019",
  dueDate: "11/03/2019",
  details: [
    {
      id: 1,
      name: "Premium Support",
      quantity: 1,
      rate: 9,
      amount: 89,
    },
    {
      id: 2,
      name: "Metror Dashboard",
      quantity: 3,
      rate: 100,
      amount: 90,
    },
    {
      id: 3,
      name: "Parts for service",
      quantity: 2,
      rate: 10,
      amount: 60,
    },
  ],
};
