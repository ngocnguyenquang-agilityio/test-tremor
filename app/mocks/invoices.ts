import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    id: 804322,
    createdAt: "March, 21, 2021",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 8908543,
    createdAt: "Apr, 22, 2022",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 415645,
    createdAt: "May, 23, 2023",
    invoicePrefix: "MS",
    totalCost: 180,
  },
  {
    id: 103578,
    createdAt: "Jun, 24, 2024",
    invoicePrefix: "QW",
    totalCost: 180,
  },
  {
    id: 126749,
    createdAt: "Jul, 25, 2025",
    invoicePrefix: "RV",
    totalCost: 180,
  },
];
