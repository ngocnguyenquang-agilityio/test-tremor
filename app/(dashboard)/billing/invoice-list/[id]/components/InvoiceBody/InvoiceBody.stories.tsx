import type { Meta, StoryObj } from "@storybook/react";

// Components
import InvoiceBody from "./InvoiceBody";

// Constants
import { INVOICE_DATA } from "@/mocks";

const meta = {
  title: "Components/InvoiceBody",
  component: InvoiceBody,
  tags: ["autodocs"],
} as Meta<typeof InvoiceBody>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <InvoiceBody
      id={INVOICE_DATA.id}
      createdAt={INVOICE_DATA.createdAt}
      dueAt={INVOICE_DATA.dueAt}
      products={INVOICE_DATA.products}
      totalCost={INVOICE_DATA.totalCost}
    />
  ),
};
