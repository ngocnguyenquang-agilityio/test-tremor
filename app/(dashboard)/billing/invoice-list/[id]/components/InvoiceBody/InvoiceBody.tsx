"use client";

// Libs
import { Flex } from "@tremor/react";

// Components
import InfoInvoiceBody from "./InfoInvoiceBody/InfoInvoiceBody";
import TableInvoice from "./TableInvoice/TableInvoice";

// Types
import { TInvoice } from "@/types";

// Styles
import "@/styles/invoice.css";

const InvoiceBody = ({
  id,
  createdAt,
  dueAt,
  products,
  totalCost,
}: TInvoice) => (
  // <Flex className="invoice-body flex-col items-end">
  <div className="block">
    <InfoInvoiceBody id={id} createdAt={createdAt} dueAt={dueAt} />
    <TableInvoice details={products} totalCost={totalCost} />
  </div>
  // </Flex>
);

export default InvoiceBody;
