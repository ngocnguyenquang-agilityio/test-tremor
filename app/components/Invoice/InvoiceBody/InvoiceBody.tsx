"use client";

// Libs
import { Flex } from "@tremor/react";

// Components
import InfoInvoiceBody from "./InfoInvoiceBody/InfoInvoiceBody";
import TableInvoiceBody from "./TableInvoiceBody/TableInvoiceBody";

// Types
import { TInvoice } from "@/types";

// Styles
import "@/styles/invoice.css";

const InvoiceBody = ({ id, created_at, dueDate, details }: TInvoice) => (
  <Flex className="invoice-body flex-col items-end">
    <InfoInvoiceBody id={id} created_at={created_at} dueDate={dueDate} />
    <TableInvoiceBody details={details} />
  </Flex>
);

export default InvoiceBody;
