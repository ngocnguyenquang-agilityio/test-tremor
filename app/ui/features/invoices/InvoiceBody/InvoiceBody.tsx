"use client";

// Libs
import { Flex } from "@tremor/react";

// Components
const InfoInvoiceBody = dynamic(
  () => import("../InfoInvoiceBody/InfoInvoiceBody"),
);
const TableInvoice = dynamic(() => import("../TableInvoice/TableInvoice"));

// Types
import { TInvoice } from "@/types";

// Styles
import "@/styles/invoice.css";
import dynamic from "next/dynamic";

const InvoiceBody = ({
  id,
  createdAt,
  dueAt,
  products,
  totalCost,
}: TInvoice) => (
  <Flex className="invoice-body flex-col items-end">
    <InfoInvoiceBody id={id} createdAt={createdAt} dueAt={dueAt} />
    <TableInvoice details={products} totalCost={totalCost} />
  </Flex>
);

export default InvoiceBody;
