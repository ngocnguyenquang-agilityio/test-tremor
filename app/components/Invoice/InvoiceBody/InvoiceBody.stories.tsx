import type { Meta, StoryObj } from "@storybook/react";

// Components
import InvoiceBody from "./InvoiceBody";

// Constants
import { INVOICE_DATA } from "@/mocks";

const meta = {
  title: "Components/InvoiceBody",
  component: InvoiceBody,
  tags: ["autodocs"],
} as Meta<typeof InvoiceBody>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <InvoiceBody
      id={INVOICE_DATA.id}
      createdAt={INVOICE_DATA.createdAt}
      dueDate={INVOICE_DATA.dueDate}
      details={INVOICE_DATA.details}
    />
  ),
};
