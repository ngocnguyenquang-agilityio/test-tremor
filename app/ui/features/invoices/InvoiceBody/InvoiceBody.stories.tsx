import type { Meta, StoryObj } from "@storybook/react";

// Components
import InvoiceBody from "./InvoiceBody";
import { Card } from "@tremor/react";

// Constants
import { INVOICE_DATA } from "@/mocks";

const meta = {
  title: "Components/Billing/Invoices/InvoiceBody",
  component: InvoiceBody,
  tags: ["autodocs"],
  argTypes: {
    id: { description: "Id of invoice detail body" },
    createdAt: { description: "Create date of invoice detail body" },
    dueAt: { description: "Due date of invoice detail body" },
    products: { description: "Products of invoice detail body" },
    totalCost: { description: "Total cost of invoice detail body" },
  },
} as Meta<typeof InvoiceBody>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <Card className="w-full dark:bg-dark_blue p-6 ring-0 rounded-xl shadow-md">
      <InvoiceBody
        id={INVOICE_DATA.id}
        createdAt={INVOICE_DATA.createdAt}
        dueAt={INVOICE_DATA.dueAt}
        products={INVOICE_DATA.products}
        totalCost={INVOICE_DATA.totalCost}
      />
    </Card>
  ),
};
