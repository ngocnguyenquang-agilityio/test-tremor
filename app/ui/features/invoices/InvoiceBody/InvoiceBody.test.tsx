import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

// Components
import InvoiceBody from "./InvoiceBody";

// Constants
import { INVOICE_DATA } from "@/mocks";

jest.mock("next/navigation", () => ({
  usePathname: jest.fn().mockReturnValue("/mocked-path"),
  useRouter: jest.fn().mockReturnValue({
    replace: jest.fn(),
  }),
  useSearchParams: jest.fn().mockReturnValue(new URLSearchParams()),
}));

describe("InvoiceBody Testing", () => {
  const invoice = {
    id: INVOICE_DATA.id,
    createdAt: INVOICE_DATA.createdAt,
    dueAt: INVOICE_DATA.dueAt,
    products: INVOICE_DATA.products,
    totalCost: INVOICE_DATA.totalCost,
  };
  it("should match snapshot", () => {
    const component = render(<InvoiceBody {...invoice} />);
    expect(component).toMatchSnapshot();
  });

  it("renders invoice body with correct information", () => {
    const { getByText } = render(<InvoiceBody {...invoice} />);

    expect(getByText("#230220")).toBeInTheDocument();
    expect(getByText("03/06/2019")).toBeInTheDocument();
    expect(getByText("03/11/2019")).toBeInTheDocument();
  });

  it("renders table with invoice details", () => {
    const { getByText } = render(<InvoiceBody {...invoice} />);

    expect(getByText(INVOICE_DATA.products[0].productName)).toBeInTheDocument();
    expect(getByText(INVOICE_DATA.products[0].quantity)).toBeInTheDocument();
  });
});
