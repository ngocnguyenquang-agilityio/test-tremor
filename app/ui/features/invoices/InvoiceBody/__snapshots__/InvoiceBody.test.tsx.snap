// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceBody Testing should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tremor-Flex-root flex w-full justify-between invoice-body flex-col items-end"
      >
        <div
          class="tremor-Flex-root flex w-full justify-between mb-7 flex-col md:flex-row items-start md:items-center"
        >
          <div
            class="w-full mb-5 md:mb-0"
          >
            <p
              class="text-base font-normal dark:text-lighter text-secondary dark:print:text-secondary"
            >
              Invoice no
            </p>
            <a
              class="text-tremor-primary text-primary dark:text-white leading-relaxed font-bold tracking-wide no-underline dark:print:text-primary"
              href="#"
            >
              #
              230220
            </a>
          </div>
          <div
            class="w-full"
          >
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col md:flex-row mb-1 md:mb-0"
            >
              <p
                class="w-full mb-1 md:mb-0 text-left md:text-end text-base font-normal dark:text-lighter text-secondary dark:print:text-secondary"
              >
                Invoice date:
              </p>
              <p
                class="w-full text-left md:text-end text-base text-primary font-semibold dark:text-white dark:print:text-primary"
              >
                03/06/2019
              </p>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col md:flex-row"
            >
              <p
                class="w-full mb-1 md:mb-0 text-left md:text-end text-base font-normal dark:text-lighter text-secondary dark:print:text-secondary"
              >
                Due date:
              </p>
              <p
                class="w-full text-left md:text-end text-base text-primary font-semibold dark:text-white dark:print:text-primary"
              >
                03/11/2019
              </p>
            </div>
          </div>
        </div>
        <div
          class="min-h-[500px] flex justify-center items-center"
          role="status"
        >
          <svg
            aria-hidden="true"
            class="w-8 h-8 text-transparent animate-[spin_0.4s_linear_infinite] dark:text-gray-600 fill-river-bed-500 dark:fill-white"
            fill="none"
            viewBox="0 0 22 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 3C11 2.44772 11.4477 2 12 2C17.5228 2 22 6.47715 22 12C22 14.7175 20.9147 17.1835 19.1562 18.9849C18.7704 19.3801 18.1373 19.3878 17.7421 19.002C17.3469 18.6162 17.3392 17.983 17.725 17.5879C19.1336 16.1449 20 14.1746 20 12C20 7.58172 16.4183 4 12 4C11.4477 4 11 3.55228 11 3Z"
              fill="currentFill"
            />
          </svg>
          <span
            class="sr-only"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tremor-Flex-root flex w-full justify-between invoice-body flex-col items-end"
    >
      <div
        class="tremor-Flex-root flex w-full justify-between mb-7 flex-col md:flex-row items-start md:items-center"
      >
        <div
          class="w-full mb-5 md:mb-0"
        >
          <p
            class="text-base font-normal dark:text-lighter text-secondary dark:print:text-secondary"
          >
            Invoice no
          </p>
          <a
            class="text-tremor-primary text-primary dark:text-white leading-relaxed font-bold tracking-wide no-underline dark:print:text-primary"
            href="#"
          >
            #
            230220
          </a>
        </div>
        <div
          class="w-full"
        >
          <div
            class="tremor-Flex-root flex w-full justify-between items-center flex-col md:flex-row mb-1 md:mb-0"
          >
            <p
              class="w-full mb-1 md:mb-0 text-left md:text-end text-base font-normal dark:text-lighter text-secondary dark:print:text-secondary"
            >
              Invoice date:
            </p>
            <p
              class="w-full text-left md:text-end text-base text-primary font-semibold dark:text-white dark:print:text-primary"
            >
              03/06/2019
            </p>
          </div>
          <div
            class="tremor-Flex-root flex w-full justify-between items-center flex-col md:flex-row"
          >
            <p
              class="w-full mb-1 md:mb-0 text-left md:text-end text-base font-normal dark:text-lighter text-secondary dark:print:text-secondary"
            >
              Due date:
            </p>
            <p
              class="w-full text-left md:text-end text-base text-primary font-semibold dark:text-white dark:print:text-primary"
            >
              03/11/2019
            </p>
          </div>
        </div>
      </div>
      <div
        class="min-h-[500px] flex justify-center items-center"
        role="status"
      >
        <svg
          aria-hidden="true"
          class="w-8 h-8 text-transparent animate-[spin_0.4s_linear_infinite] dark:text-gray-600 fill-river-bed-500 dark:fill-white"
          fill="none"
          viewBox="0 0 22 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3C11 2.44772 11.4477 2 12 2C17.5228 2 22 6.47715 22 12C22 14.7175 20.9147 17.1835 19.1562 18.9849C18.7704 19.3801 18.1373 19.3878 17.7421 19.002C17.3469 18.6162 17.3392 17.983 17.725 17.5879C19.1336 16.1449 20 14.1746 20 12C20 7.58172 16.4183 4 12 4C11.4477 4 11 3.55228 11 3Z"
            fill="currentFill"
          />
        </svg>
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
